Sources/ServiceLifecycle/GracefulShutdown.swift

```diff
@@ -115,7 +115,7 @@ enum ValueOrGracefulShutdown<T: Sendable>: Sendable {
 /// Cancels the closure when a graceful shutdown was triggered.
 ///
 /// - Parameter operation: The actual operation.
-public func cancelOnGracefulShutdown<T: Sendable>(_ operation: @Sendable @escaping () async throws -> T) async rethrows -> T? {
+public func cancelOnGracefulShutdown<T: Sendable>(_ operation: @Sendable @escaping () async throws -> T) async rethrows -> T {
```

Comment:
So we gotta be a bit careful here since this is a breaking change. Can we deprecate this method and introduce the same one. This one can call the new one to avoid duplication.

Reply:
I was also concerned about that, but couldn't find a suitable solution.

We could provide an overload method without an ambiguous error as follows:

```swift
@available(*, deprecated)
@_disfavoredOverload
public func cancelOnGracefulShutdown<T: Sendable>(_ operation: @Sendable @escaping () async throws -> T) async rethrows -> T? {
    ...
}

public func cancelOnGracefulShutdown<T: Sendable>(_ operation: @Sendable @escaping () async throws -> T) async rethrows -> T {
    ...
}
```

However, this overload does not address the source-breaking issue.
Since `T` can be assigned to `T?`, the new method will be used automatically.

```swift
// OK
let value: Int? = await cancelOnGracefulShutdown {
    return 42
}
```

Nevertheless, if there were a use case like the following, it would break existing code, and we couldn't resolve it with an overload:

```swift
// Source breaking!
let value = await cancelOnGracefulShutdown {
    return 42
}
if let value {
    ...
}
```

Aside from providing an overload, the only other option would be to use a different name, but I didn’t think it was appropriate to rename this function.

I just can't think of a good approach.

Reply:
Yep that's completely true. We gotta introduce a new name here sadly. Let's deprecate the current method and introduce a new one called `cancelWhenGracefulShutdown`. It isn't great but it gets us moving forward.

Reply:
OK, I have implemented `cancelWhenGracefulShutdown`.